lib/mjsonwp/routes.js

```diff
@@ -455,7 +455,55 @@ for (let v of _.values(METHOD_MAP)) {
   }
 }
 
+const RE_ESCAPE = /[\-\[\]{}()+?.,\\\^$|#\s]/g;
+const RE_PARAM = /([:*])(\w+)/g;
+
+class Route {
+  constructor (route) {
+    this.paramNames = [];
+
+    let reStr = route.replace(RE_ESCAPE, "\\$&");
+    reStr = reStr.replace(RE_PARAM, (_, mode, name) => {
+      this.paramNames.push(name);
+      return mode === ":" ? "([^/]*)" : "(.*)";
+    });
+    this.routeRegexp = new RegExp(`^${reStr}$`);
+  }
+
+  parse (url) {
+    let matches = url.match(this.routeRegexp);
+    if (!matches) return;
+    let i = 0;
+    let params = {};
+    while (i < this.paramNames.length) {
+      const paramName = this.paramNames[i++];
+      params[paramName] = matches[i];
+    }
+    return params;
+  }
+}
+
+function routeToCommandName (endpoint, method) {
+  let dstEndpoint = null;
+  for (let currentEndpoint of _.keys(METHOD_MAP)) {
+    const route = new Route(currentEndpoint);
+    // endpoint should already include leading slash
+    // we don't care about the actual session id for matching
+    if (route.parse(`/wd/hub/session/x${endpoint}`) || route.parse(`/wd/hub${endpoint}`) || route.parse(endpoint)) {
```

Comment:
Then you need another condition that handles needing `/wd/hub/session` appended. Or your test is testing the wrong endpoint value (`'/8b8da8ad-c8c1-42f5-be5a-35ff592beceb/timeouts/implicit_wait'`)